ember-file-upload/src/system/data-transfer-wrapper.ts

```diff
@@ -43,6 +43,69 @@ export default class DataTransferWrapper {
     return this.files.length ? this.files : this.items;
   }
 
+  async getFilesAndDirectories() {
+    const files: File[] = [];
+
+    const readEntry = async (entry: FileSystemEntry): Promise<File> => {
+      return new Promise((resolve, reject) => {
+        if (entry.isFile) {
+          (entry as FileSystemFileEntry).file((fileEntry: File) => {
+            resolve(fileEntry);
+          });
+        } else {
+          reject('Directory contains nested directories');
+        }
+      });
+    };
+
+    const readAllFilesInDirectory = (item: DataTransferItem): Promise<File[]> =>
+      new Promise((resolve) => {
+        (item.webkitGetAsEntry() as FileSystemDirectoryEntry)
```

Comment:
The [advice from MDN](https://developer.mozilla.org/en-US/docs/Web/API/DataTransferItem/webkitGetAsEntry) about this method is:

> Note: This function is implemented as webkitGetAsEntry() in non-WebKit browsers including Firefox at this time; it may be renamed to getAsEntry() in the future, so you should code defensively, looking for both.

I'd suggest that we wrap this function and attempt to call `getAsEntry` and then `webkitGetAsEntry`. Hopefully that will be supported in future without us needing to make a change

Reply:
Wrapped it in a local `getEntry` function. Typescript does complain because `getAsEntry` is not (yet) in the lib definition. Tried to circumvent this by extending the interface to `FutureProofDataTransferItem` , not sure if that's the cleanest solution.